### DISCUSSION TITLE
Restoring my pocketbase data for docker usage on VPS

### DISCUSSION THREAD

**Discussion by furkankarayel** [pocketbase/pocketbase#4501] [Q&A] [answered]
Hello guys, I have got a backup of my latest pb data and would like to restore it on a vps instance where I run pb in docker.
I don't really get it how to make it restore based on a Dockerfile.

This is my Dockerfile I currently use:

FROM alpine:latest

RUN apk add --no-cache  \
    bash \
    unzip \
    ca-certificates

ARG PB_VERSION=0.12.0
ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_arm64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb
RUN mkdir /pb/pb_data

EXPOSE 8090

CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8090", "--dir='/root/pb_data'"]

Feel free to correct me if you would do something in a different or smoother way.

Another question would be; What would be important to do and don't when hosting pb on my own vps?

**Comment by ganigeorgiev** [comment]
You can import the backup from the _Admin UI > Settings > Backups > Upload backup_.

If you want to restore it manually, you need to upload the archive's `pb_data` to the location where your Docker volume directory is located.

For syncing collections configuration changes (aka. without data) between environments I recommend using Migrations ([Go](https://pocketbase.io/docs/go-migrations/) or [JS](https://pocketbase.io/docs/js-migrations/)).

**Reply by furkankarayel** [reply]
I don't have the option called Backups in Settings...

**Reply by ganigeorgiev** [reply, marked as answer]
Ah, that is because you are using a very old PocketBase version (_if you don't overwrite the `ARG PB_VERSION` instructions based on your post it seems that you are using v0.12.0; option to upload backups was added in v0.18.0; the latest version is v0.22.3_).

In any case if you are planning to upgrade, I recommend reviewing the CHANGELOGs as big jumps between versions are not guarantee to always work:

- for <= v0.15.0 - https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG_8_15.md
- for >= v0.16.0 - https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md

**Reply by furkankarayel** [reply]
Thanks for the tip I appreciate it. I have updated to the last version because I assume that pockethost uses it too. I have imported the collections config and tried to execute the migration, in the cli I get this error:

Error: Failed to apply migration 1709745947_created_carousels.js: constraint failed: UNIQUE constraint failed: _collections.name (2067)

**Reply by ganigeorgiev** [reply]
What migration? What collections config? Sorry but this is really confusing and I don't think I'll be able to help you further.
My recommendation from earlier remains. The warnings in the CHANGELOGs and the docs are there for a reason.

**Reply by furkankarayel** [reply]
Sorry for confusing you, I thought it's necessary to import the collections (Settings>Import collections) before the backup can be imported but that was a wrong assumption because everything is in the backup already.
The Backup has been imported successfully, so just changing the version to the last one released was helpful.

Thank you very much, appreciate it.
